csharp/src/Apache.Arrow/Memory/LongSpan.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Apache.Arrow.Memory
+{
+    internal readonly struct LongSpan<T>
+    {
+        private readonly LongMemory<T> _memory;
```

Comment:
It might be a good idea to look at how the existing Span type and other corresponding types are implemented in .NET. Eg. if you go to the docs for the Span type (https://learn.microsoft.com/en-us/dotnet/api/system.span-1?view=net-8.0) there's a "Source" link that takes you to https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Private.CoreLib/src/System/Span.cs

There you can see Span holds:
```
        internal readonly ref T _reference;
        private readonly int _length;
```

I imagine that LongSpan should then look like:
```
        internal readonly ref T _reference;
        private readonly long _length;
```